core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+        annotationLine.setSource("TitleBlockLine");
+
+        // addColumnsToLine(annotationLine, representation, numCols); start
+        List<DAnnotation> annotationCols = new ArrayList<DAnnotation>();
+        for (int j = 0; j < numCols; j++) {
+          DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+          annotationCol.setSource("TB_" + i + "_" + j);
+          annotationCol.getDetails().put(NAME, "Name");
+          annotationCol.getDetails().put(CONTENT, "feature:name");
+          annotationCols.add(annotationCol);
+          representation.getEAnnotations().add(annotationCol);
+        }
+        annotationLine.getReferences().addAll(annotationCols);
+        representation.getEAnnotations().add(annotationLine);
+        // stop
+
+        annotationLines.add(annotationLine);
+      }
+
+      if (!elementView.equals(diagram)) {
+        annotation.getReferences().add(((DDiagramElement) elementView).getTarget());
+        annotation.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+      }
+      annotation.getReferences().addAll(annotationLines);
+      representation.getEAnnotations().add(annotation);
+    }
+  }
+
+  public void createTitleBlockLine(EObject titleBlock, EObject diagram) {
+    if (!isDiagramLevel(titleBlock)) {
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        DAnnotation annotationLine = DescriptionFactory.eINSTANCE.createDAnnotation();
+        annotationLine.setSource("TitleBlockLine");
+        if (titleBlock instanceof DAnnotation) {
+          int numCols = getNumColumns((DAnnotation) titleBlock);
+          if (numCols > 0) {
+            ((DAnnotation) titleBlock).getReferences().add(annotationLine);
+            addColumnsToLine(annotationLine, representation, numCols, getNumLines((DAnnotation) titleBlock));
+          }
+        }
+      }
+    }
+  }
+
+  public void createTitleBlockColumn(EObject titleBlock, EObject diagram) {
+    if (!isDiagramLevel(titleBlock)) {
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        if (titleBlock instanceof DAnnotation) {
+          int numLines = ((DAnnotation) titleBlock).getReferences().size();
+          List<EObject> lines = ((DAnnotation) titleBlock).getReferences();
+          for (int i = 0; i < numLines; i++) {
+            DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+            annotationCol.setSource("TitleBlockLineCol");
+            if (lines.get(i) instanceof DAnnotation) {
+              ((DAnnotation) (lines.get(i))).getReferences().add(annotationCol);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private boolean isDiagramLevel(EObject titleBlock) {

```

Comment:
rename maybe to isDiagramTileBlock

Reply:
Removed it, as there was already a isDiagramTitleBlock method.